ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/PartitionExport.java

```diff
@@ -115,11 +119,19 @@ void write(final ReplicationSpec forReplicationSpec) throws InterruptedException
                   .export(forReplicationSpec);
           LOG.debug("Thread: {}, finish partition dump {}", threadName, partitionName);
         } catch (Exception e) {
-          throw new RuntimeException("Error while export of data files", e);
+          throw new RuntimeException(e.getMessage(), e);
         }
-      });
+      }));
     }
     consumer.shutdown();
+    for (Future<?> future : futures) {
+      try {
+        future.get();
+      } catch (Exception e) {
```

Comment:
It is better to catch only RuntimeException as otherwise it may suppress others.

Reply:
it was suppressed earlier ..now we are getting the exceptions and throwing it explicitly